democracy_club/apps/core/tests/test_embed_code.py

```diff
@@ -0,0 +1,16 @@
+from django.test import TestCase
+from django.urls import reverse
+
+
+class TestEmbedCode(TestCase):
+    def test_language(self):
+        """assert when a user chooses cy as a language from the
+        language options, the embed code includes cy as an attribute"""
+        url = reverse("projects:polling_embed_code")
+        form_input = {
+            "name": "language",
+            "id": "cy",
+            "value": "cy",
+        }
+        response = self.client.get(url, data=form_input)
+        self.assertTrue('data-language="cy"', response.content)
```

Comment:
I don't think this test is actually testing what you think it is....the form on the site doesn't accept data form GET or POST params, so I don't think the form data is getting passed to anything. Also, the test runner won't run JavaScript, so I don't think it's possible to test that it's working in this way.

You can prove this by adding 

```py
response = self.client.get(url)
self.assertTrue('data-language="cy"', response.content)
```

Right below the `url=` line.

Because you're using `assertTrue`, you're asserting that `bool('data-language="cy"')` evaluates to `True`. This is always true. The second argument is the message to show if the expression isn't true, so is never getting evaluated.

Because testing this is hard, I suggest just deleting the test case for the time being.

Reply:
To test this properly, we'll need to set up some frontend testing systems. More on that later today!

Reply:
Thank you. I tried this all sorts of ways, but nothing targeted the test the way I wanted.  I almost added Jest but then thought it was worth checking first, esp as there isn't much opportunity for user input on Website.

Reply:
Nice catch!